### PR TITLE
Custom hour of day label formatter

### PR DESCRIPTION
**Pull request by kennethj**
These changes add the features to custom the hour column. Currently, we can only set the TextStyle with the hour format of `HH:mm` and column width set at `48.0`. This update adds the ability to set custom time formatting and column width.

- [x] Adds theme property for formatting the hour of the day label.
- [x] Adds theme property to set the width of hour column

### Example
```dart
final theme = TimetableThemeData(
  formatHour: (time) => '${time.hourOf12HourClock} ${time.hourOfDay > 11 ? "PM" : "AM"}',
  hourColumnWidth: 40,
);
```

### Checklist

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

#### Format hour
| Android  | iOS  |
| :------------: | :------------: |
| Screen Shot 2020-07-19 at 4 16 10 PM | Screen Shot 2020-07-19 at 4 19 12 PM |

#### Set column width 
| Larger  | Zero width  |
| :------------: | :------------: |
| Screen Shot 2020-07-20 at 9 56 19 AM | Screen Shot 2020-07-20 at 9 56 32 AM|

### CODE REVIEW

File: example/.vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "name": "Flutter",
+            "request": "launch",
+            "type": "dart"
+        }
+    ]
+}
```
Comment (AndroidNils) [inline example/.vscode/launch.json:13]: I think the `launch.json` shouldn't be in this PR, right?

Comment (kennethj) [reply example/.vscode/launch.json:13]: You are right. I'll remove it.

**Comment by TheMkrage** [comment]
Is there any plan for this still? This is something we would love to make use of! Thank you

**Comment by IoanaAlexandru** [comment]
@TheMkrage while waiting for a PR to get approved, you can always just specify it in your dependencies:
```
timetable:
    git: https://github.com/iu-uits-es/timetable.git
```

**Comment by TatsuUkraine** [comment]
@JonasWanke any timeline when this feature will be merged?

**Comment by TatsuUkraine** [comment]
nvm) this solution still doesn't fit for me))) I need more customizable time labels)))

**Comment by Ahmadre** [comment]
@kennethj could you please check the conflicts? And @JonasWanke Any feedback to this?

**Comment by JonasWanke** [comment]
Thanks for your efforts! Due to the major refactor tracked in #17, I, unfortunately, can't directly use this PR, but am incorporating customizable time labels supporting custom widgets (addressing #41, #58, and #61).

**Pull request closed without merging**